### PR TITLE
NewWithBracesFixer - fix handling more neighbor tokens

### LINKED ISSUE #1863: new_with_braces fixer generating invalid code
Here is the diff:
```
      -        if ($this->startDate instanceof DateTime && new DateTime > $this->startDate) {
      +        if ($this->startDate instanceof DateTime && new DateTime > $this->startDate()) {
```

### PR DESCRIPTION
**Pull request by keradus**
Closes #1863

### CODE REVIEW

File: Symfony/CS/Tests/Fixer/Symfony/NewWithBracesFixerTest.php [hunk 1]
```diff
@@ -123,6 +123,30 @@ public function provideStandardCases()
...
+                '<?php
+                    if ($this->startDate instanceof DateTime && new DateTime > $this->startDate) {}
+                    if ($this->startDate instanceof DateTime && new DateTime >= $this->startDate) {}
+                    if ($this->startDate instanceof DateTime && new DateTime < $this->startDate) {}
+                    if ($this->startDate instanceof DateTime && new DateTime <= $this->startDate) {}
+                    if ($this->startDate instanceof DateTime && new DateTime == $this->startDate) {}
+                    if ($this->startDate instanceof DateTime && new DateTime != $this->startDate) {}
+                    if ($this->startDate instanceof DateTime && new DateTime <> $this->startDate) {}
+                    if ($this->startDate instanceof DateTime && new DateTime === $this->startDate) {}
+                    if ($this->startDate instanceof DateTime && new DateTime !== $this->startDate) {}
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/NewWithBracesFixerTest.php]: it may be worth to add things like `+` and `>>`

**Comment by GrahamCampbell** [comment]
Looks like the spaceship operator is missing from here too?

**Comment by keradus** [comment]
done @GrahamCampbell

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by SpacePossum** [comment]
request to add this to the tests, it makes no sense but is valid PHP ;)

```
<?php
$a = new \stdClass ? $this->a : $this->b;
var_dump($a);
die;
```

**Comment by gharlan** [comment]
`php
foreach (new Collection as $x {}

$a = [(string) new Foo => 1];

```

**Comment by SpacePossum** [comment]
we might want to prep. for `??=` as well (https://github.com/php/php-src/pull/1795) ;)

**Comment by gharlan** [comment]
could give us a valid example with `??=` (as next token)? :)

**Comment by GrahamCampbell** [comment]
> we might want to prep. for ??= as well (php/php-src#1795) ;)

Nah. Not yet. It's not merged in yet, and we're not testing on PHP 7.1 yet. We should just make a new issue with PHP 7.1 things todo, like we have for 7.0.

**Comment by keradus** [comment]
@GrahamCampbell please could you create a ticket with 7.1 wishlist and update ticket for 7.0 if needed ?

**Pull request merged**